nsot/models.py

```diff
@@ -2161,6 +2167,46 @@ def to_dict(self):
             'resource': resource,
         }
 
+    @property
+    def diff(self):
+        """
+        Return the diff of the JSON representation of the cached copy of a
+        Resource with its current instance
+        """
+        if self.event == 'Create':
+            old = ''
+        else:
+            # Get the Change just ahead of _this_ change because that has the
+            # state of the Resource before this Change occurred.
+            # TODO(nickpegg): Get rid of this if we change the behavior of
+            # Change to store the previous version of the object
+            old_change = Change.objects.filter(
+                change_at__lt=self.change_at,
+                resource_id=self.resource_id,
+                resource_name=self.resource_name
+            ).order_by(
+                '-change_at'
+            )[0]
+            old = json.dumps(old_change._resource, indent=2, sort_keys=True)
+
+        if self.event == 'Delete':
+            current = ''
+        else:
+            resource = globals()[self.resource_name]
```

Comment:
This line kind of scares me. If we want to use something canonical, I'd recommend:

```
from django.apps import apps

# in method:
resource = apps.get_model('nsot', self.resource_name)
```

And if we don't want to hard-code the `app_label` (`nsot`) we can get that from `self._meta.app_label`, e.g.

```
resource = apps.get_model(self._meta.app_label, self.resource_name)
```